### PR TITLE
feat: save playbook run as config change

### PR DESCRIPTION
**Pull request by adityathebe**
resolves: https://github.com/flanksource/mission-control/issues/1238

image
image

### CODE REVIEW

**Comment by adityathebe** [comment]
image

File: playbook/playbook.go [hunk 1]
```diff
@@ -96,9 +97,60 @@ func validateAndSavePlaybookRun(ctx context.Context, playbook *models.Playbook,
...
+		Severity:         models.SeverityInfo,
+		ConfigID:         playbook.ID.String(),
+		ChangeType:       string(run.Status),
+		Source:           "Playbook",
+		Summary:          fmt.Sprintf("Playbook run %s", run.Status),
+	}
+
+	switch run.Status {
+	case models.PlaybookRunStatusScheduled:
+		change.Severity = models.SeverityLow
```
Comment (moshloop) [inline playbook/playbook.go:126]:
```suggestion
		change.Severity = models.SeverityInfo
```

File: playbook/playbook.go [hunk 2]
```diff
@@ -96,9 +97,60 @@ func validateAndSavePlaybookRun(ctx context.Context, playbook *models.Playbook,
...
+		detailsJSON, err := json.Marshal(details)
+		if err != nil {
+			return fmt.Errorf("error marshaling playbook details into config changes: %w", err)
+		}
+		change.Details = detailsJSON
+
+	case models.PlaybookRunStatusRunning:
+		change.ChangeType = "started"
+		change.Summary = "Playbook run started"
+		change.Severity = models.SeverityHigh
```
Comment (moshloop) [inline playbook/playbook.go:142]:
```suggestion
		change.Severity = models.SeverityInfo
```

**Comment by adityathebe** [comment]
image

File: playbook/playbook.go [hunk 3]
```diff
@@ -96,9 +96,56 @@ func validateAndSavePlaybookRun(ctx context.Context, playbook *models.Playbook,
...
+		ExternalChangeId: uuid.NewString(),
+		Severity:         models.SeverityInfo,
+		ConfigID:         run.ConfigID.String(),
+		ChangeType:       string(run.Status),
+		Source:           "Playbook",
+		Summary:          fmt.Sprintf("Playbook: %s", playbook.Name),
+	}
+
+	switch run.Status {
+	case models.PlaybookRunStatusScheduled:
```
Comment (moshloop) [inline playbook/playbook.go:121]: Can the Change type be `PlaybookScheduled` ?

Comment (adityathebe) [reply playbook/playbook.go:123]: image

File: playbook/playbook.go [hunk 4]
```diff
@@ -96,9 +98,56 @@ func validateAndSavePlaybookRun(ctx context.Context, playbook *models.Playbook,
...
+			return fmt.Errorf("error marshaling playbook details into config changes: %w", err)
+		}
+		change.Details = detailsJSON
+
+	case models.PlaybookRunStatusRunning:
+		change.ChangeType = "PlaybookStarted"
+		change.Severity = models.SeverityInfo
+
+	case models.PlaybookRunStatusCompleted:
+		change.Severity = models.SeverityLow
```
Comment (moshloop) [inline playbook/playbook.go:142]:
```suggestion
                 change.ChangeType = "PlaybookCompleted"
		change.Severity = models.SeverityLow
```

File: playbook/playbook.go [hunk 5]
```diff
@@ -96,9 +98,56 @@ func validateAndSavePlaybookRun(ctx context.Context, playbook *models.Playbook,
...
+
+	case models.PlaybookRunStatusFailed:
+		change.Severity = models.SeverityHigh
```
Comment (moshloop) [inline playbook/playbook.go:145]:
```suggestion
		change.Severity = models.SeverityHigh
		change.ChangeType = "PlaybookFailed"
```

File: playbook/playbook.go [hunk 6]
```diff
@@ -96,9 +98,56 @@ func validateAndSavePlaybookRun(ctx context.Context, playbook *models.Playbook,
...
+		Severity:         models.SeverityInfo,
+		ConfigID:         run.ConfigID.String(),
+		ChangeType:       fmt.Sprintf("Playbook%s", cases.Title(language.English).String(string(run.Status))),
+		Source:           "Playbook",
+		Summary:          fmt.Sprintf("Playbook: %s", playbook.Name),
+	}
+
+	switch run.Status {
+	case models.PlaybookRunStatusScheduled:
+		change.Severity = models.SeverityInfo
```
Comment (moshloop) [inline playbook/playbook.go:124]:
```suggestion
		change.Severity = models.SeverityInfo
		change.ChangeType = "PlaybookScheduled"
```

Comment (adityathebe) [reply playbook/playbook.go:124]: @moshloop these change types were already set above.

**Pull request merged**